docs/designs/concepts.md

```diff
@@ -32,60 +32,41 @@ Let's make a short summary, following is all the core concepts of ElasticDL incl
 
 ## Message Representation
 
+There is a [tensor](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.proto) proto message defined in TensorFlow, which meets our needs. We could reuse it directly.
+
 We introduce an `IndexedSlices` proto message to represent the concatenated embedding vectors pulled from PS, and the concatenated embedding vectors of gradient waiting to be pushed to PS.
 
 The definition of `elasticdl.proto`:
 
 ```proto
-enum ElementType {
-  // Not a legal value for DataType. Used to indicate a DataType field
-  // has not been set.
-  DT_INVALID = 0;
-
-  DT_INT8 = 1;
-  DT_INT16 = 2;
-  DT_INT32 = 3;
-  DT_INT64 = 4;
-  DT_FLOAT16 = 5;
-  DT_FLOAT32 = 6;
-  DT_FLOAT64 = 7;
-  DT_BOOL = 8;
-}
-
-message Tensor {
-  repeated int64 dims = 1;
-  bytes content = 2;
-  ElementType dtype = 3;
-}
+import "tensorflow/tensorflow/core/framework/tensor.proto"
 
 message IndexedSlices {
-  Tensor concat_embedding_vecs = 1;
+  tensorflow.Tensor concat_tensors = 1;
   repeated int64 ids = 2;
 }
 
 message Model {
   int32 version = 1; // model updated times
-  map<string, Tensor> dense_parameters = 2;
-  map<string, IndexedSlices> indexed_slices = 3;
+  map<string, tensorflow.Tensor> dense_parameters = 2;
+  map<string, IndexedSlices> embedding_tables = 3;
 }
 ```
 
 For in-memory part, we introduce an `EmbeddingTable` data structure.
 
 ```go
-import "elasticdl.org/elasticdl/pkg/proto"
-
 type EmbeddingTable struct {
     Name            string
     Dim             int64
     Initializer     string
-    EmbeddingVector map[int64]*proto.Tensor
+    EmbeddingVector map[int64]*Tensor
 }
 
 type Model struct {
     Version           int32
     InitStatus        bool
-    DenseParameters   map[string]*proto.Tensor
+    DenseParameters   map[string]*Tensor
```

Comment:
Do we need a `tensorflow.` prefix before Tensor

Reply:
Good point.  Fix it.